storage/series/series.go

```diff
@@ -440,6 +442,31 @@ func (s *dbSeries) mergeBlock(
 	return nil
 }
 
+func (s *dbSeries) Update(blk block.DatabaseBlock) error {
+	s.Lock()
+	defer s.Unlock()
+
+	// ensure series is bootstraped before allowing any updates
+	if s.bs != bootstrapped {
+		return errSeriesNotBootstrapped
+	}
+
+	// we only accept updates to non-buffered data
+	// ensure buffer is past provided time
+	min, _ := s.buffer.MinMax()
+	if !blk.StartTime().Before(min) {
```

Comment:
Awesome! One thing to point out, I had an off by one in the time ranges I was operating upon - this [line](https://github.com/m3db/m3db/pull/214/files/4bf93c374fdd55cb46960d13334e205abc6eb500#diff-97d06de2227f251da9f03d6e77a03927R536). 

Will change it to `end   = retention.FlushTimeEnd(r.rtopts, startTime).Add(-blockSize)` in the next version